### ISSUE TITLE
Docker build failed on M1 Mac

### ISSUE THREAD

**Issue by vimesh-shah** [dotnet/eShopOnContainers#1771]
```
=> CACHED [eshop/basket.api:linux-latest build 36/41] RUN dotnet restore "eShopOnContainers-ServicesAndWebApps.sln"                                                                      0.0s
 => CACHED [eshop/basket.api:linux-latest build 37/41] COPY . .                                                                                                                           0.0s
 => CACHED [eshop/basket.api:linux-latest build 38/41] WORKDIR /src/Services/Basket/Basket.API                                                                                            0.0s
 => CACHED [eshop/basket.api:linux-latest build 39/41] RUN export PROTOBUF_TOOLS_OS=linux                                                                                                 0.0s
 => CACHED [eshop/basket.api:linux-latest build 40/41] RUN export PROTOBUF_TOOLS_CPU=x64                                                                                                  0.0s
 => ERROR [eshop/basket.api:linux-latest build 41/41] RUN dotnet publish --no-restore -c Release -o /app                                                                                  2.6s
------
 > [eshop/basket.api:linux-latest build 41/41] RUN dotnet publish --no-restore -c Release -o /app:
#51 0.414 Microsoft (R) Build Engine version 16.11.1+3e40a09f8 for .NET
#51 0.414 Copyright (C) Microsoft Corporation. All rights reserved.
#51 0.414
#51 2.155   EventBus -> /src/BuildingBlocks/EventBus/EventBus/bin/Release/net5.0/EventBus.dll
#51 2.446   EventBusRabbitMQ -> /src/BuildingBlocks/EventBus/EventBusRabbitMQ/bin/Release/net5.0/EventBusRabbitMQ.dll
#51 2.466   EventBusServiceBus -> /src/BuildingBlocks/EventBus/EventBusServiceBus/bin/Release/net5.0/EventBusServiceBus.dll
#51 2.566 /root/.nuget/packages/grpc.tools/2.34.0/build/_protobuf/Google.Protobuf.Tools.targets(84,5): error : Google.Protobuf.Tools cannot determine host OS and CPU. [/src/Services/Basket/Basket.API/Basket.API.csproj]
#51 2.566 /root/.nuget/packages/grpc.tools/2.34.0/build/_protobuf/Google.Protobuf.Tools.targets(84,5): error : Use environment variables PROTOBUF_TOOLS_OS={linux|macosx|windows} and PROTOBUF_TOOLS_CPU={x86|x64} to try the closest match to your system. [/src/Services/Basket/Basket.API/Basket.API.csproj]
#51 2.566 /root/.nuget/packages/grpc.tools/2.34.0/build/_protobuf/Google.Protobuf.Tools.targets(84,5): error : You may also set PROTOBUF_PROTOC to specify full path to the host-provided compiler (v3.5+ is required). [/src/Services/Basket/Basket.API/Basket.API.csproj]
------
failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c dotnet publish --no-restore -c Release -o /app]: exit code: 1
```

**Comment by sughosneo** [comment]
Hi @vimesh-shah, Thank you for reaching out.

I just tried to build the solution in MacOS it seems to be working fine. Did you take the latest from the `dev` branch?
Also, could you please try the below steps?

- To do clean up 

```text

docker stop $(docker ps -aq)
docker rm $(docker ps -aq)
docker volume rm $(docker volume ls -q)

```

> !IMPORTANT: Above commands will clean up all running containers and it's data in your MacOS/Workstation.

- Enable `docker.for.linux.localhost` hostname for the docker in the following file. 

https://github.com/dotnet-architecture/eShopOnContainers/blob/46219957ef763305030a0924d039fe4a25831b06/src/.env#L16

- To run the app 
- Go to the `/src/` directory

```text
docker-compose up sqldata
```
Once it's ready to accept the new connection, then you can bring up the other container by using the following command from a separate terminal window.

```text
docker-compose up
```

**Comment by SofianeBraneci** [comment]
Hello @sughosneo,
I just tried the steps above, and the issue is not resolved.
I'm on macOS BigSur, M1 chip.

**Comment by nishanil** [comment]
@SofianeBraneci I don't think we can support ARM64 until we have all dependant images natively supporting it. Even if we put in efforts right now, we can only use Intel emulations, resulting in frequent crashes. You may try building them using the instructions listed in [Docker Desktop for Apple Silicon](https://docs.docker.com/desktop/mac/apple-silicon/) and let us know if it worked.

**Comment by yadazula** [comment]
Hi. 

1. The gRPC issue that @vimesh-shah reported can be resolved by upgrading the "Grpc.Core" package to v2.37.0 or above. More detail on this: https://github.com/grpc/grpc/issues/23336

2. I applied this gRPC package upgrade but faced another issue with 'sqldata' image as SQL Server isn’t supported on the ARM architecture. There are some workarounds like using Azure SQL Edge but I didn't try it though. Other than that, I didn't face any problems.

**Comment by nishanil** [comment]
@yadazula Thanks for reporting it.

**Comment by Mart112358** [comment]
Hello,

Based on the comments in this thread and playing around with my M1 pro, I think I was able to make everything work.

I was able to authenticate with the demo user and create orders for both the SPA and MVC apps.

I also ran the unit/functionnal tests and everything passed.

What I did:

- Docker: Replace MSSQL image by Azure SQL Edge
- Docker: Update envoy proxy to 1.16.0 (first version with ARM64 images) 
- Update Grpc* references to the latest version

See the changes here: https://github.com/Mart112358/eShopOnContainers/tree/fix/1771
